core/modules/wsched/ChunkTaskList.cc

```diff
@@ -0,0 +1,387 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+// Class header
+#include "wsched/ChunkTaskList.h"
+
+// System headers
+#include "global/Bug.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.wsched.ChunkTaskList");
+}
+
+namespace lsst {
+namespace qserv {
+namespace wsched {
+
+
+void ChunkTaskList::queTask(wbase::Task::Ptr const& task) {
+    std::lock_guard<std::mutex> lg(_mapMx);
+    int chunkId = task->getChunkId();
+    auto iter = _chunkMap.find(chunkId);
+    if (iter == _chunkMap.end()) {
+        iter = _insertChunkTask(chunkId);
+    }
```

Comment:
This check could be performed in_insertChunkTask to avoid test on res.second and simplify alittle bit the code?